File: src/main/org/nlogo/prim/etc/_left.java
// (C) 2012 Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.agent.Turtle;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;

public final strictfp class _left
    extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax
        (new int[]{Syntax.NumberType()},
            "-T--", true);
  }

  @Override
  public void perform(final Context context)
      throws LogoException {
    perform_1(context, argEvalDoubleValue(context, 0));
  }

  public void perform_1(final Context context, double delta) {
    ((Turtle) context.agent).turnRight(-delta);
    context.ip = next;
  }
}

File: src/main/org/nlogo/prim/etc/_left.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Turtle
import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Command, Context }

class _left extends Command {
  override def syntax =
    Syntax.commandSyntax(
      Array(Syntax.NumberType), "-T--", true)
  override def perform(context: Context) {
    perform_1(context, argEvalDoubleValue(context, 0));
  }
  def perform_1(context: Context, delta: Double) {
    context.agent.asInstanceOf[Turtle].turnRight(-delta)
    context.ip = next
  }
}
